### ISSUE TITLE
BigQuery API Doesn't support TIME data type with Subsecond

### ISSUE THREAD

**Issue by rendybjunior** [googleapis/google-cloud-python#6141]
OS type:  `macOS High Sierra 10.13.6 (17G65)`
Python version: `Python 2.7.15` (I'm working with apache beam, so inevitable to use 2.7)
Library: `google-cloud-bigquery==1.5.0`

Stacktrace
```
Traceback (most recent call last):
  File "test_dict.py", line 13, in <module>
    for row in results:
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/api_core/page_iterator.py", line 200, in _items_iter
    for item in page:
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/api_core/page_iterator.py", line 125, in next
    result = self._item_to_value(self._parent, item)
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/cloud/bigquery/table.py", line 1338, in _item_to_row
    return Row(_helpers._row_tuple_from_json(resource, iterator.schema),
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/cloud/bigquery/_helpers.py", line 212, in _row_tuple_from_json
    row_data.append(converter(cell['v'], field))
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/cloud/bigquery/_helpers.py", line 146, in _time_from_json
    return _time_from_iso8601_time_naive(value)
  File "/Users/username/virtualenv/virtualenvname/lib/python2.7/site-packages/google/cloud/_helpers.py", line 261, in _time_from_iso8601_time_naive
    return datetime.datetime.strptime(value, '%H:%M:%S').time()
  File "/usr/local/Cellar/python@2/2.7.15/Frameworks/Python.framework/Versions/2.7/lib/python2.7/_strptime.py", line 335, in _strptime
    data_string[found.end():])
ValueError: unconverted data remains: .450000
```

Steps to reproduce
1. Create table with TIME data type
2. Insert sample data
```sql
INSERT INTO `project.dataset.table` (field_with_time_type) VALUES ("12:30:00.45'')
```
3. Query 
```python
from google.cloud import bigquery
client = bigquery.Client()
query_job = client.query("""SELECT field_with_time_type FROM `project.dataset.table`""")
results = query_job.result()
for row in results:
    print(row)
```

**Comment by rendybjunior** [comment]
Seems like it has been solved for datetime and timestamp, but not with time

**Comment by tseaver** [comment]
I've got to confess that I am puzzled by the usecase for microsecond-precision, dateless times, but the [`Time` type does support it](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#time-type).

**Comment by tseaver** [comment]
Fixed in #5753, which was included in the `google-cloud-bigquery 1.5.1` release.  Please try:

```bash
$ pip install --upgrade google-cloud-bigquery
```
